util/proxy/proxy.go

```diff
@@ -0,0 +1,50 @@
+package proxy
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"os/exec"
+	"strings"
+)
+
+// AddEnvIfAbsent adds the proxy URL as an env variable if absent
+func AddEnvIfAbsent(cmd *exec.Cmd, proxy string) []string {
+	envs := []string{}
+	if proxy == "" {
+		return cmd.Env
+	}
+	// remove the existing proxy env variable if present
+	for i, env := range cmd.Env {
+		proxyEnv := strings.ToLower(env)
+		if strings.HasPrefix(proxyEnv, "http_proxy") || strings.HasPrefix(proxyEnv, "https_proxy") {
+			continue
+		}
+		envs = append(envs, cmd.Env[i])
+	}
+	if strings.HasPrefix(proxy, "https://") {
+		envs = append(envs, httpsProxy(proxy))
+	} else {
+		envs = append(envs, httpProxy(proxy))
+	}
```

Comment:
If _proxy_ is set to non-empty string, we should set both `http_proxy` and `https_proxy` in the environment here.

The logic with `http_proxy` and `https_proxy` variables is that the first part of the variable's name indicates the _destination_ protocol, not the protocol the proxy itself is accessed with.

For example, if you want to access a destination `https://foo.bar` through proxy `http://proxy:3128`, the value of variable `https_proxy` is used. Likewise, if you want to access `http://foo.bar` through proxy `https://proxy:3128`, the value of variable `http_proxy` is used.

Reply:
Thanks for the explanation! Will set both `http_proxy` and `https_proxy` in case of non-empty string